src/udev/udev-node.c

```diff
@@ -296,8 +296,11 @@ static int node_permissions_apply(sd_device *dev, bool apply_mac,
         else
                 mode |= S_IFCHR;
 
-        if (lstat(devnode, &stats) < 0)
+        if (lstat(devnode, &stats) < 0) {
+                if (errno == ENOENT)
+                        return 0; /* this is necessarily racey, so ignore missing the device */
                 return log_device_debug_errno(dev, errno, "cannot stat() node %s: %m", devnode);
+        }
```

Comment:
looks good, but what about the operation below? shouldn't they all just fail silently on ENOENT? i.e. the chmod_and_chown() and so on?

Reply:
Hmm??? On ENOENT, this return with 0, so `chmod_and_chown()` is not called.

Reply:
well, but what happens if during the lstat() the device is still there, but then during the chmod() it is gone? I mean, we never know when precisely the device disappears, could be any time, hence we should handle it on all operations?

(Alternatively we could of course operate on an open(O_PATH) fd instead, so that removal doesn't invalidate the object for us)

Reply:
I see. I understand. Now, ENOENT in `chmod_and_chown()` or later operations are also ignored. PTAL.